set.mm

```diff
@@ -412491,6 +412575,135 @@ about equation (2) in section I.1 in [Bollobas] p. 4.  (Contributed by
+  walk (of length n) is represented by the following sequence:
+
+  p(0) e(f(1)) p(1) e(f(2)) ... p(n-1) e(f(n)) p(n).
+
+  Based on this definition of ` Walks `, the class of all walks, more
+  restrictive constructs are defined:
+
+  * ` Trails `: A "walk is called a _trail_ if all its edges are distinct.",
+    see Definition of [Bollobas] p. 5, i.e., f(i) =/= f(j) if i =/= j.
+
+  * ` Paths `: A _path_ is a walk whose vertices except the first and the last
+    vertex are distinct, i.e., p(i) =/= p(j) if i =/= j, except i = 0 and
+    j = n.
+
+  * ` SPaths ` (simple paths): A _simple path_ "is a walk with distinct
+    vertices.", see Notation of [Bollobas] p. 5, i.e., p(i) =/= p(j)
+    if i =/= j.
+
+  * ` ClWalks ` (closed walks): A walk whose endvertices coincide is called a
+    _closed walk_, i.e.,  p(0) = p(n).
+
+  * ` Circuits `: "A trail whose endvertices coincide (a closed trail) is
+    called a _circuit_." (see Definition of [Bollobas] p. 5), i.e.,
+    f(i) =/= f(j) if i =/= j and p(0) = p(n).
+
+  * ` Cycles `: A path whose endvertices coincide (a closed path) is called a
+    _cycle-, i.e., i.e. p(i) =/= p(j) if i =/= j, except i = 0 and j = n, and
+    p(0) = p(n).
+
+  * ` EulerPaths ` (Eulerian paths): An _Eulerian path_ is "a trail containing
+    all edges" (see definition in [Bollobas] p. 16), i.e.,  f(i) =/= f(j) if
+    i =/= j and for all edges e(x) there is an 1 <= i <= n with e(x) = e(f(i)).
+
+  * Eulerian circuit: An _Eulerian circuit_ (called _Euler tour_ in the
+    definition in [Diestel] p. 22) is "a circuit in a graph containing all the
+    edges" (see definition in [Bollobas] p. 16), i.e.,  f(i) =/= f(j) if
+    i =/= j, p(0) = p(n) and for all edges e(x) there is an 1 <= i <= n with
+    e(x) = e(f(i)).
+
+  Hierarchy of all kinds of walks (apply ~ ssriv and ~ elopabran to the
+  mentioned theorems to obtain the following subset relationships, as available
+  for ~ clwlkiswlk , see ~ clwlkwlk and ~ clwlkswks ):
+
+  * Trails are walks ( ~ trliswlk ): ` ( Trails `` G ) C_ ( Walks `` G ) `
+
+  * Paths are trails ( ~ pthistrl ): ` ( Paths `` G ) C_ ( Trails `` G ) `
+
+  * Simple paths are paths ( ~ spthispth ):
+    ` ( SPaths `` G ) C_ ( Paths `` G ) `
+
+  * Closed walks are walks ( ~ clwlkiswlk ):
+    ` ( ClWalks `` G ) C_ ( Walks `` G ) `
+
+  * Circuits are closed walks ( ~ crctisclwlk ):
+    ` ( Circuits `` G ) C_ ( ClWalks `` G ) `
+
+  * Circuits are trails ( ~ crctistrl ):
+    ` ( Circuits `` G ) C_ ( Trails `` G ) `
+
+  * Cycles are paths ( ~ cyclispth ): ` ( Cycles `` G ) C_ ( Paths `` G ) `
+
+  * Cycles are circuits ( ~ cycliscrct ):
+    ` ( Cycles `` G ) C_ ( Circuits `` G ) `
+
+  * (Non trivial) cycles are not simple paths ( ~ cyclnspth ):
+    ` ( F =/= (/) -> ( F ( Cycles `` G ) P -> -. F ( SPaths `` G ) P ) ) `
+
+  * Eulerian paths are trails ( ~ eupthistrl ):
+    ` ( EulerPaths `` G ) C_ ( Trails `` G ) `
+
```

Comment:
Is an Eulerian path a path ?

Reply:
not necessarily, see above.

Reply:
Since this is a red herring, it may be worth repeating here: "Note, however, that an Eulerian path need not be a path."